staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/handler.go

```diff
@@ -65,9 +68,55 @@ type ResourceManager interface {
 	// Thread-Safe
 	SetGroups([]apidiscoveryv2beta1.APIGroupDiscovery)
 
+	// Returns the same resource manager using a different source
+	// The source is used to decide how to de-duplicate groups.
+	// DefaultSource always takes precedence. Otherwise, the precedence between
+	// two sources is not defined.
+	WithSource(source string) ResourceManager
```

Comment:
I get that this way you don't have to change the interface, but wouldn't it be good to force all callers to pass a source string?

Reply:
Personally I like how it is currently organized, but am happy to change if you feel strongly about it